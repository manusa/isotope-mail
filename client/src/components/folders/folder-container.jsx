import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import Spinner from '../spinner/spinner';
import FolderList from './folder-list';
import FolderRenameDialog from './folder-rename-dialog';
import styles from './folder-container.scss';
import mainCss from '../../styles/main.scss';

export const FolderContainer =
  ({activeRequests, folderList}) => (
    <nav className={`${mainCss['mdc-list']}`}>
      <Spinner visible={activeRequests > 0 && folderList.length === 0}
        canvasClassName={styles.spinnerCanvas} />
      <FolderList folderList={folderList}/>
      <FolderRenameDialog />
    </nav>
  );

FolderContainer.propTypes = {
  activeRequests: PropTypes.number.isRequired,
  folderList: PropTypes.array.isRequired
};

const mapStateToProps = state => ({
  activeRequests: state.folders.activeRequests,
  folderList: state.folders.items
});

export default connect(mapStateToProps)(FolderContainer);
