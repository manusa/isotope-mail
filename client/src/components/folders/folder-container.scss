.spinnerCanvas {
  width: 24px;
}
