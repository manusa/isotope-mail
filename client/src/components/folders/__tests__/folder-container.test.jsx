import React from 'react';
import {shallow} from 'enzyme';
import {FolderContainer} from '../folder-container';

describe('FolderContainer component test suite', () => {
  test('Snapshot render, should render FolderContainer', () => {
    const {activeRequests, folderList, selectedFolder} = {
      activeRequests: 0, folderList: [{folderList: []}], selectedFolder: {folderId: '1337-3'}};
    const folderContainer = shallow(
      <FolderContainer activeRequests={activeRequests} folderList={folderList}
        selectedFolder={selectedFolder}/>);
    expect(folderContainer).toMatchSnapshot();
  });
});
