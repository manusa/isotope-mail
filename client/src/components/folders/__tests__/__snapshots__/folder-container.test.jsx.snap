// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderContainer component test suite Snapshot render, should render FolderContainer 1`] = `
<nav
  className="mdc-list"
>
  <Spinner
    canvasClassName="spinnerCanvas"
    className=""
    pathClassName=""
    visible={false}
  />
  <Connect(FolderListClass)
    folderList={
      Array [
        Object {
          "folderList": Array [],
        },
      ]
    }
  />
  <Connect(Translate(FolderRenameDialog)) />
</nav>
`;
